python/eos/pyhf_likelihood.py

```diff
@@ -44,31 +44,52 @@ def __init__(self, cache, workspace, parameter_map={}):
         self.model = self.workspace.model()
         self.data = np.array(self.workspace.data(self.model, include_auxdata=False))
 
-        self.varied_parameters = []
-        for pn, v, bounds in zip(self.model.config.par_names,
-                                 self.model.config.suggested_init(),
-                                 self.model.config.suggested_bounds()):
-
-            if pn in parameter_map:
-                qn = eos.QualifiedName(parameter_map[pn])
+        self._observables = []
+        for parameter_name, init, bounds in zip(
+            self.model.config.par_names,
+            self.model.config.suggested_init(),
+            self.model.config.suggested_bounds()
+            ):
+
+            kinematics = eos.Kinematics()
+            options = eos.Options()
+
+            if parameter_name in parameter_map:
+                if isinstance(parameter_map[parameter_name], str):
+                    qualified_name = eos.QualifiedName(
+                        parameter_map[parameter_name]
+                        )
+                elif isinstance(parameter_map[parameter_name], dict):
+                    qualified_name = eos.QualifiedName(
+                        parameter_map[parameter_name]['name']
+                        )
+                    kinematics = eos.Kinematics(parameter_map[parameter_name].get('kinematics', {}))
+                    options = eos.Options(parameter_map[parameter_name].get('options', {}))
+                else:
+                    raise ValueError('parameter_map values must be either strings or dictionaries.')
             else:
-                qn = eos.QualifiedName('pyhf::' + pn)
+                qualified_name = eos.QualifiedName('pyhf::' + parameter_name)
 
-            if self.parameters.has(qn):
-                p = self.parameters[qn]
+            if self.parameters.has(qualified_name):
```

Comment:
The logic here is off. You have 3 cases:
1. ``qualified_name`` corresponds to an existing parameter;
2. ``qualified_name`` corresponds to an observable name;
3. ``qualified_name`` is unknown to EOS.

You are currently handling case 1 in line 74, and both cases 2 and 3 in lines 76...83. That is definitely buggy. I am unsure if that causes the warning you see, though.

Reply:
Ah I see. This indeed solves the problem. Before it created a new parameter with the same name as the observable, even if it existed. Hence, the warning. Should be fixed now.

Reply:
Brilliant!